rca/navigation/models.py

```diff
@@ -7,8 +9,50 @@
 from wagtail.core.fields import StreamField
 
 
+def url_or_relative_url_validator(value):
+    if value.startswith("/"):
+        value = "http://rca.ac.uk" + value
+    return validators.URLValidator()(value)
+
+
+class URLOrRelativeURLFormField(forms.URLField):
+    default_validators = [url_or_relative_url_validator]
+
+    def to_python(self, value):
+        value = super().to_python(value)
+        if value:
+            if value.startswith("http:///"):
+                value = value[7:]
```

Comment:
What is this? I have no idea what you are doing here.

Reply:
Without this, the validation won't pass, the return value for `/schools/` would be `http://schools/` so we need to strip the http from it to return `/schools/`

Reply:
IMO you could rely more on the URL manipulation capabilities of the standard library.

If it's `http:///schools/`, then this should work. Won't work for `http://schools/`.

```python
from urllib.parse import urlsplit, urlunsplit
    def to_python(self, value):
        value = super().to_python(value)
        if value:
            url_fields = list(urlsplit(value))
            # If netloc (domain) is empty, delete the scheme
            if not url_fields[1]:
                url_fields[0] = ''
             value = urlunsplit(url_fields)
```

Reference:
https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlsplit
https://github.com/django/django/blob/ca0d50f34ad14be2eca3067011c659ef0e66a0bc/django/forms/fields.py#L665-L694

Reply:
:+1: thanks Tomasz